File: 백준/Silver/2839. 설탕 배달/설탕 배달.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        List<Integer> anw = new ArrayList<>();

        for (int i = 0; i <= n / 3; i++) {
            for (int j = 0; j <= n / 5; j++) {
                if (3 * i + 5 * j == n) {
                    anw.add(i + j);
                }
            }
        }

        if (!anw.isEmpty()) {
            Collections.sort(anw);
            System.out.println(anw.get(0));
        } else {
            System.out.println(-1);
        }
    }
}

File: 백준/Silver/2839. 설탕 배달/설탕 배달.kt
import java.util.Scanner
fun main(args: Array<String>) = with(Scanner(System.`in`)){
    var n = nextInt()
    val anw = mutableListOf<Int>()
    (0..n / 3).map { i -> (0..n / 5).forEach { j -> if (3 * i + 5 * j == n) anw.add(i + j) } }
    println(if (anw.isNotEmpty()) anw.sorted().first() else -1)
}